jina/orchestrate/deployments/__init__.py

```diff
@@ -459,52 +471,27 @@ def num_pods(self) -> int:
     def __eq__(self, other: 'BaseDeployment'):
         return self.num_pods == other.num_pods and self.name == other.name
 
-    def activate(self):
-        """
-        Activate all worker pods in this deployment by registering them with the head
-        """
-        if self.head_pod is not None:
-            for shard_id in self.pod_args['pods']:
-                for pod_idx, pod_args in enumerate(self.pod_args['pods'][shard_id]):
-                    worker_host = self.get_worker_host(
-                        pod_args, self.shards[shard_id]._pods[pod_idx], self.head_pod
-                    )
-                    GrpcConnectionPool.activate_worker_sync(
-                        worker_host,
-                        int(pod_args.port),
-                        self.head_pod.runtime_ctrl_address,
-                        shard_id,
-                    )
-
     @staticmethod
-    def get_worker_host(pod_args, pod, head_pod):
+    def get_worker_host(pod_args, pod_is_container, head_is_container):
         """
         Check if the current pod and head are both containerized on the same host
         If so __docker_host__ needs to be advertised as the worker's address to the head
 
         :param pod_args: arguments of the worker pod
-        :param pod: the worker pod
-        :param head_pod: head pod communicating with the worker pod
-        :return: host to use in activate messages
+        :param pod_is_container: boolean specifying if pod is to be run in container
+        :param head_is_container: boolean specifying if head pod is to be run in container
+        :return: host to pass in connection list of the head
         """
         # Check if the current pod and head are both containerized on the same host
         # If so __docker_host__ needs to be advertised as the worker's address to the head
         worker_host = (
             __docker_host__
-            if Deployment._is_container_to_container(pod, head_pod)
+            if (pod_is_container and (head_is_container or in_docker()))
```

Comment:
The naming of `_is_docker` and `in_docker` is very similar. From what I understand the one checks a provided deployment, while the other checks the current process. Could this be clarified in the naming?